geopandas/tools/sjoin.py

```diff
@@ -86,6 +86,26 @@ def sjoin(
     """
     _basic_checks(left_df, right_df, how, lsuffix, rsuffix)
 
+    box_left_gdf = left_df.total_bounds
+    box_right_gdf = right_df.total_bounds
+
+    if not (
+        (
+            (box_left_gdf[0] <= box_right_gdf[2])
+            and (box_right_gdf[0] <= box_left_gdf[2])
+        )
+        and (
+            (box_left_gdf[1] <= box_right_gdf[3])
+            and (box_right_gdf[1] <= box_left_gdf[3])
+        )
+    ):
+        copy_df = left_df.copy(deep=True)
```

Comment:
Do we need `deep=True` here?

Reply:
I got used to do it, but indeed, seems to be the default.